### PR TITLE
Manual Testing Bugs fixes

### LINKED ISSUE #182: [PE-D][Tester A] Adding event with year "0000" leads to error.
The program does not add events with the year "0000". No mention of not allowing year 0000 is mentioned in the User Guide. The error message claims this is due to an invalid date-time format, despite "0000-12-12 12:00" following the specified constraints: 

1. YYYY-MM-DD HH:mm
2. MM-DD has to be between 01-01 and 12-31, and HH:mm has to be between 00:00 and 23:59.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Dri-water/ped#3

### LINKED ISSUE #185: [PE-D][Tester D] Invalid dates are accepted and modified automatically
Invalid dates such as 30 February and 31 February are accepted in the program. Furthermore, these dates are then modified to be 29 February when listing the corresponding events.

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: ZiliaAJY/ped#2

### LINKED ISSUE #208: [PE-D][Tester D] Able to filter an invalid date
I am able to filter with an invalid date, and there are results returned for it as well. Event with date February 29 is returned when i filter for an event with date February 30.

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: ZiliaAJY/ped#9

### LINKED ISSUE #248: SortCommand Bug
Random parameters does not have exception handling

### PR DESCRIPTION
**Pull request by KuanHsienn**
Fixes #248 
Fixes #208 
Fixes #185 
Fixes #182

### CODE REVIEW

File: src/main/java/seedu/manager/parser/Parser.java [hunk 1]
```diff
@@ -930,15 +926,21 @@ private boolean toMarkItem(String status) throws InvalidCommandException, IOExce
      * @throws InvalidCommandException if the command is missing required parts or has an invalid format
      */
     private Command parseCopyCommand(String input, String[] commandParts) throws InvalidCommandException {
```
Comment (LTK-1606) [inline src/main/java/seedu/manager/parser/Parser.java:928]: maybe can add the check for duplicate flags here as well and make sure they don't use the '>'  more than once?

File: src/main/java/seedu/manager/parser/Parser.java [hunk 2]
```diff
@@ -960,17 +962,25 @@ private Command parseCopyCommand(String input, String[] commandParts) throws Inv
      */
     private Command parseSortCommand(String input, String[] commandParts) throws InvalidCommandException {
```
Comment (LTK-1606) [inline src/main/java/seedu/manager/parser/Parser.java:963]: Maybe should add the duplicate flag check for '-by' here too

File: src/main/java/seedu/manager/parser/Parser.java [hunk 3]
```diff
@@ -992,20 +1002,26 @@ private Command parseSortCommand(String input, String[] commandParts) throws Inv
...
+
+        Pattern pattern = Pattern.compile(FILTER_REGEX);
+        Matcher matcher = pattern.matcher(input);
+
+        if (matcher.matches()) {
 
-        Set<String> validFlags = Set.of(EVENT_FLAG, "-d", "-t", "-x", "-u");
-        if (validFlags.contains(commandParts[1].trim().toLowerCase())) {
-            return new FilterCommand(commandParts[1].trim().toLowerCase(), inputParts[1].trim());
+            if (matcher.group(1).isBlank()) {
```
Comment (LTK-1606) [inline src/main/java/seedu/manager/parser/Parser.java:1013]: matcher.group(1) should be the filter description keywords. There isn't a matcher.group(2)

File: src/main/java/seedu/manager/command/MenuCommand.java [hunk 4]
```diff
@@ -25,7 +25,7 @@ public class MenuCommand extends Command {
             mark -p PARTICIPANT -e EVENT -s STATUS: Mark a participant as present or absent.
             mark -m ITEM -e EVENT -s STATUS: Mark an item as accounted or unaccounted.
             copy FROM_EVENT > TO_EVENT: Copies participant list from one event to another.
-            sort -by KEYWORD: Sorts events by name/time/priority.
+            sort -e EVENT -by KEYWORD: Sorts events by name/time/priority.
```
Comment (LTK-1606) [inline src/main/java/seedu/manager/command/MenuCommand.java:28]: Can I clarify what this means? User needs to specify the events?

**Review by LTK-1606 — COMMENTED** [review]
Overall looks good, just a few things to take note of

File: src/main/java/seedu/manager/parser/Parser.java (hunk 3 above)
Comment (KuanHsienn) [reply src/main/java/seedu/manager/parser/Parser.java:1042]: i used group one as the flag cause got multiple types of flag that i need to pass to the filter command

Comment (LTK-1606) [reply src/main/java/seedu/manager/parser/Parser.java:1042]: As in I do not think matcher.group(2) exists? Like there should not be any value inside it

Comment (LTK-1606) [reply src/main/java/seedu/manager/parser/Parser.java:1042]: ok my bad

**Comment by LTK-1606** [comment]
LGTM

**Comment by LTK-1606** [comment]
Where did you catch the invalid dates like 29/30/31 Feb? I couldn't see it in the code

**Comment by KuanHsienn** [comment]
> Where did you catch the invalid dates like 29/30/31 Feb? I couldn't see it in the code

As in? it is similar to the previous datetimeparser, they just throw an exception if its invalid but datetimeparser library will rectify dates. so in this case im using a new library simpledataformat which does not rectify the date and the exception thrown when there is invalid date is parseexception.

**Comment by LTK-1606** [comment]
I didnt see the new lib. so it will throw exception for invalid dates including 29/30/31 feb, then the event won't get added at all right?

**Comment by KuanHsienn** [comment]
> I didnt see the new lib. so it will throw exception for invalid dates including 29/30/31 feb, then the event won't get added at all right?

Yupp

**Comment by LTK-1606** [comment]
LGTM, can merge alr

**Pull request merged**